src/main/java/com/librato/disco/DiscoClient.java

```diff
@@ -69,33 +70,20 @@ public boolean isStarted() {
     }
 
     public List<Node<T>> getAllNodes() {
-        List<ChildData> data = cache.getCurrentData();
-        if (data.isEmpty()) {
-            return Collections.emptyList();
-        }
-        List<Node<T>> nodes = new ArrayList<>(data.size());
-        for (ChildData child : data) {
-            nodes.add(toNode(child));
-        }
-        return nodes;
+        return cache.getCurrentData().stream().map(this::toNode).collect(Collectors.toList());
```

Comment:
That was my first instinct, but that's one more function lookup... I imagine the JVM will optimize it away, but I can't bring myself to do it.